### PR TITLE
Add `--ssl` options to `ember test`

### PR DESCRIPTION
**Pull request by nathanhammond**
`testem` supports SSL configuration but we don't expose or forward configuration of that via Ember CLI.

### CODE REVIEW

**Comment by nathanhammond** [comment]
@Turbo87 I've added descriptions to the SSL-related options for both `ember serve` and `ember test` in a separate commit. (Didn't exist previously.)

File: lib/commands/test.js [hunk 1]
```diff
@@ -29,6 +29,9 @@ module.exports = Command.extend({
     { name: 'launch',      type: String,  default: false,                            description: 'A comma separated list of browsers to launch for tests.' },
     { name: 'reporter',    type: String,                            aliases: ['r'],  description: 'Test reporter to use [tap|dot|xunit] (default: tap)' },
     { name: 'silent',      type: Boolean, default: false,                            description: 'Suppress any output except for the test report' },
+    { name: 'ssl',         type: Boolean, default: false,                            description: 'Set to true to configure testem to run the test suite using SSL.' },
```
Comment (nathanhammond) [inline lib/commands/test.js]: This is the only behavior change.

File: lib/tasks/test.js [hunk 2]
```diff
@@ -63,6 +63,13 @@ class TestTask extends Task {
       query_params: options.queryString,
       /* eslint-enable camelcase */
     };
+
+    if (options.ssl) {
```
Comment (nathanhammond) [inline lib/tasks/test.js]: This is the only logic change.

File: tests/unit/tasks/test-test.js [hunk 3]
```diff
@@ -34,7 +38,33 @@ describe('test task test', function() {
...
+
+    subject.run({
+      host: 'greatwebsite.com',
+      port: 123324,
+      reporter: 'xunit',
+      outputPath: 'blerpy-derpy',
+      testemDebug: 'testem.log',
+      testPage: 'http://my/test/page',
+      configFile: 'custom-testem-config.json',
+      ssl: false,
```
Comment (nathanhammond) [inline tests/unit/tasks/test-test.js]: We don't receive default options in our test mocks, so we need to provide them. This tests one branch of the new bit of logic.

File: tests/unit/tasks/test-test.js [hunk 4]
```diff
@@ -34,7 +38,33 @@ describe('test task test', function() {
...
+      sslKey: 'ssl/server.key',
+      sslCert: 'ssl/server.cert',
+    });
+  });
+
+  it('supports conditionally passing SSL configuration forward', function() {
```
Comment (nathanhammond) [inline tests/unit/tasks/test-test.js]: This tests the other branch of the new bit of logic.

**Comment by Turbo87** [comment]
@nathanhammond can you rebase? CI (at least Travis) should be fixed now

**Comment by nathanhammond** [comment]
Thanks, @Turbo87! (I was out of town.)

**Pull request merged**